--- a/src/main/java/org/bukkit/command/PluginCommand.java
+++ b/src/main/java/org/bukkit/command/PluginCommand.java
@@ -32,12 +_,14 @@
     @Override
     public boolean execute(@NotNull CommandSender sender, @NotNull String commandLabel, @NotNull String[] args) {
         boolean success = false;
+        gg.pufferfish.pufferfish.sentry.SentryContext.setPluginContext(owningPlugin); // Pufferfork - taken from pufferfish-gg/pufferfish/pull/64
 
         if (!owningPlugin.isEnabled()) {
             throw new CommandException("Cannot execute command '" + commandLabel + "' in plugin " + owningPlugin.getDescription().getFullName() + " - plugin is disabled.");
         }
 
         if (!testPermission(sender)) {
+            gg.pufferfish.pufferfish.sentry.SentryContext.removePluginContext(); // Pufferfork - taken from pufferfish-gg/pufferfish/pull/64
             return true;
         }
 
@@ -52,7 +_,7 @@
                 sender.sendMessage(line);
             }
         }
-
+        gg.pufferfish.pufferfish.sentry.SentryContext.removePluginContext(); // Pufferfork - taken from pufferfish-gg/pufferfish/pull/64
         return success;
     }
 
@@ -132,6 +_,7 @@
         Preconditions.checkArgument(args != null, "Arguments cannot be null");
         Preconditions.checkArgument(alias != null, "Alias cannot be null");
 
+        gg.pufferfish.pufferfish.sentry.SentryContext.setPluginContext(owningPlugin); // Pufferfork - taken from pufferfish-gg/pufferfish/pull/64
         List<String> completions = null;
         try {
             if (completer != null) {
@@ -150,6 +_,7 @@
             throw new CommandException(message.toString(), ex);
         }
 
+        gg.pufferfish.pufferfish.sentry.SentryContext.removePluginContext(); // Pufferfork - taken from pufferfish-gg/pufferfish/pull/64
         if (completions == null) {
             return super.tabComplete(sender, alias, args);
         }
