--- a/src/main/java/org/bukkit/map/MapPalette.java
+++ b/src/main/java/org/bukkit/map/MapPalette.java
@@ -394,9 +_,15 @@
         temp.getRGB(0, 0, temp.getWidth(), temp.getHeight(), pixels, 0, temp.getWidth());
 
         byte[] result = new byte[temp.getWidth() * temp.getHeight()];
+        // Pufferfish start
+        if (!gg.pufferfish.pufferfish.simd.SIMDDetection.isEnabled) {
         for (int i = 0; i < pixels.length; i++) {
             result[i] = matchColor(new Color(pixels[i], true));
         }
+        } else {
+            gg.pufferfish.pufferfish.simd.VectorMapPalette.matchColorVectorized(pixels, result);
+        }
+        // Pufferfish end   
         return result;
     }
 
@@ -511,4 +_,18 @@
         @org.jetbrains.annotations.ApiStatus.Internal // Paper
         byte matchColor(@NotNull Color color);
     }
+
+    // Pufferfork start - public access to colors
+    /**
+     * Gets the palette's color map
+     * @return The color map
+     * @apiNote Internal Use Only
+     */
+    @org.jetbrains.annotations.ApiStatus.Internal
+    @Deprecated
+    @NotNull
+    public static final Color[] getColors() {
+        return colors;
+    }
+    // Pufferfork end
 }
