--- a/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/net/minecraft/server/dedicated/DedicatedServer.java
@@ -131,6 +_,7 @@
             this.setPreventProxyConnections(properties.preventProxyConnections);
             this.setLocalIp(properties.serverIp);
         }
+        gg.pufferfish.pufferfish.PufferfishConfig.load(); // Pufferfish - Rebranding
 
         this.setPvpAllowed(properties.pvp);
         this.setFlightAllowed(properties.allowFlight);
