--- a/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/net/minecraft/server/dedicated/DedicatedServer.java
@@ -160,6 +_,7 @@
             this.setPreventProxyConnections(properties.preventProxyConnections);
             this.setLocalIp(properties.serverIp);
         }
+        gg.pufferfish.pufferfish.PufferfishConfig.load(); // Pufferfish - Rebranding
 
         // Spigot start
         this.setPlayerList(new DedicatedPlayerList(this, this.registries(), this.playerDataStorage));
