--- a/src/main/java/io/papermc/paper/ServerBuildInfo.java
+++ b/src/main/java/io/papermc/paper/ServerBuildInfo.java
@@ -19,6 +_,13 @@
      */
     Key BRAND_PAPER_ID = Key.key("papermc", "paper");
 
+    // Pufferfork start - rebrand
+    /**
+     * The brand id for Pufferfork.
+     */
+    Key BRAND_PUFFERFORK_ID = Key.key("toffikk", "pufferfork");
+    // Pufferfork end
+
     /**
      * Gets the {@code ServerBuildInfo}.
      *
