--- a/src/main/java/io/papermc/paper/ServerBuildInfo.java
+++ b/src/main/java/io/papermc/paper/ServerBuildInfo.java
@@ -19,6 +_,13 @@
      */
     Key BRAND_PAPER_ID = Key.key("papermc", "paper");
 
+    // Pufferfish start - rebrand
+    /**
+     * The brand id for Pufferfish.
+     */
+    Key BRAND_PUFFERFISH_ID = Key.key("pufferfish", "pufferfish");
+    // Pufferfish end
+
     /**
      * Gets the {@code ServerBuildInfo}.
      *
